ui/client/src/queries/common.ts

```diff
@@ -26,16 +26,13 @@ export const generatePostReq = (stringBody: string): RequestInit => {
 };
 
 export const returnResponse = async (
-  res: Response,
+  resFn: () => Promise<Response>,
   errorMsg: string,
   ignoreStatuses: number[] = []
 ) => {
+  const res = await resFn();
   if (!res.ok && !ignoreStatuses.includes(res.status)) {
     throw new Error(errorMsg);
   }
-  try {
-    return (await res.json()).result;
-  } catch {
-    return {};
-  }
+  return (await res.json()).result;
```

Comment:
The original goal of this try catch was to quietly handle cases where the value returned is not a valid JSON. Note that with this change we assume the response will always be a valid JSON.

Reply:
Very helpful to have that - I couldn't work that out from the code alone, as it seemed to be deliberately translating errors in a success payload. I thought that was the intent of the code.

A note that I didn't see a change / comments / intent described in the changes under #399, so if you have more updates to make a single approach to the different types of errors coming back to the UI true, that would be great.

The "error" categories include:
- Something that isn't JSON/RPC came back - like a `504` gateway error, which won't be JSON parsable
- An error specific to a JSON/RPC call came back, with an error code, which could be displayed to a user
- An error specific to a JSON/RPC call came back, which in that particular case the UI wants to hide from the user (like the fact there is no domain receipt on the node)
- `null` came back from a query describing a straight forward not found scenario